src/controllers/statisticController.js

```diff
@@ -106,5 +106,49 @@ module.exports = {
       await logger.error(`failed to fetch the popular item: ${err.message}`);
       res.status(500).send(err.message);
     }
+  },
+  getSucceededAndFailedPayment: async (req, res) => {
+    try {
+      const editedMonth = req.params.month > 10 ? req.params.month : `0${req.params.month}`;
+      const startString = `${req.params.year}-${editedMonth}-01T00:00:01Z`;
+      const endOfMonthDay = new Date(endOfMonth(new Date(req.params.year, req.params.month - 1)));
+      const endString = `${req.params.year}-${editedMonth}-${endOfMonthDay.getDate()}T23:59:59Z`;
+      const start = getUnixTime(new Date(startString));
+      const end = new Date(endString);
+
+      const paymentIntentsInCents = await stripeRepo.getPaymentIntentsInCents(start, end);
+      const succeededPaymentIntents = paymentIntentsInCents.data.filter(paymentIntent => paymentIntent.status === 'succeeded');
+      const failedPaymentIntents = paymentIntentsInCents.data.filter(paymentIntent => paymentIntent.status === 'failed');
+
+      const succeededPaymentCount = succeededPaymentIntents.length;
+      const failedPaymentCount = failedPaymentIntents.length;
+
+      await res.send(`Succeeded payments: ${succeededPaymentCount}\nFailed payments: ${failedPaymentCount}`);
+    } catch (err) {
+      await logger.error(`failed to fetch succeeded and failed payments: ${err.message}`);
+      res.status(500).send(err.message);
+    }
+  },
+  getSucceededAndFailedPaymentByRange: async (req, res) => {
+    try {
+      const start = new Date(req.params.start_date);
+      const end = new Date(req.params.end_date);
+
+      if (start > end) {
+        throw new Error('Invalid date range.');
+      }
+      const paymentIntents = await stripeRepo.getPaymentIntentsInCents(start.getTime() / 1000, end.getTime() / 1000);
```

Comment:
Why are you using this function which gets timestamp instead of creating a new one? The range does not contain time.. 
In addition, I think you can send the date without the .getTime() / 1000 and it will work aswell